### PR TITLE
feat(gh-action-crawling): add crawling options to action input metadata

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Details

This PR adds crawling options similar to [accessibility-insights-scan](https://www.npmjs.com/package/accessibility-insights-scan). It adds:
- url
- maxUrls
- discoveryPatterns
- inputFile
- inputUrls

tested [here](https://github.com/karanbirsingh/test-a11y-action/runs/2463027566?check_suite_focus=true) (action accepts `max-urls` in test)

##### Motivation

We want to pass these along to the crawler.

##### Context

In the CLI we do some parsing and error checking:
- https://github.com/microsoft/accessibility-insights-service/blob/main/packages/cli/src/cli.ts#L28
- https://github.com/microsoft/accessibility-insights-service/blob/main/packages/cli/src/crawler-parameters-builder.ts

We might want to expose some of this from AICrawler and reuse it. Otherwise in a follow-up PR we can reimplement some of these error checks.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -22,7 +22,20 @@ inputs:
...
     url:
-        description: 'remote site url'
+        description: 'The hosted URL to scan/crawl for accessibility issues.'
+        required: false
+    max-urls:
+        description: 'Maximum number of pages opened by crawler. The crawl will stop when this limit is reached.'
+        required: false
+        default: '100'
+    discovery-patterns:
+        description: 'List of URLs to crawl in addition to URLs discovered from crawling the provided URL, separated by space.'
```
Comment (lisli1) [inline action.yml:32]: Is this a typo? Based on https://github.com/microsoft/accessibility-insights-service/blob/main/packages/cli/src/cli.ts#L102, seems like this should read 'List of RegEx patterns to crawl in addition to the provided URL, separated by space.'

File: src/task-config.spec.ts [hunk 2]
```diff
@@ -4,90 +4,46 @@ import 'reflect-metadata';
...
+        'parses "$inputValue" as $expectedValue when provided as $inputOption',
+        ({ inputOption, getInputFunc, inputValue, expectedValue }) => {
+            actionCoreMock
+                .setup((am) => am.getInput(inputOption))
+                .returns(() => inputValue)
+                .verifiable(Times.once());
+            const retrievedOption = getInputFunc();
+            expect(retrievedOption).toStrictEqual(expectedValue);
+        },
+    );
```
Comment (lisli1) [inline src/task-config.spec.ts:46]: I like that this removes a lot of redundancy for testing each getter!

File: action.yml (hunk 1 above)
Comment (karanbirsingh) [reply action.yml]: fixed, good catch!

**Review by lisli1 — APPROVED** [review]
LGTM

**Pull request merged**